otelcol/config.go

```diff
@@ -50,13 +51,13 @@ func (cfg *Config) Validate() error {
 
 	// Currently, there is no default receiver enabled.
 	// The configuration must specify at least one receiver to be valid.
-	if len(cfg.Receivers) == 0 {
+	if !pipelines.AllowNoPipelines.IsEnabled() && len(cfg.Receivers) == 0 {
 		return errMissingReceivers
 	}
 
 	// Currently, there is no default exporter enabled.
 	// The configuration must specify at least one exporter to be valid.
-	if len(cfg.Exporters) == 0 {
+	if !pipelines.AllowNoPipelines.IsEnabled() && len(cfg.Exporters) == 0 {
 		return errMissingExporters
 	}
```

Comment:
I wasn't looking to make this change in this PR, but I think we should consider removing these checks, since validating the pipelines configuration will catch the same error.